core/libp2p/routing/router.hpp

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_ROUTER_HPP
+#define KAGOME_ROUTER_HPP
+
+#include <queue>
+
+#include "libp2p/common/peer_info.hpp"
+#include "libp2p/multi/multihash.hpp"
+
+namespace libp2p::routing {
+  /**
+   * Provides a way to find other peers in the network to establish
+   * connections with them
+   */
+  class Router {
+   public:
+    /**
+     * Find peers 'responsible' or 'closest' to a given key; explanation of
+     * these terms can be found in the articles about Kademlia DHT
+     * @param key to be searched for
+     * @return collection of found peers, sorted by 'closeness' to the key
+     */
+    virtual std::priority_queue<common::PeerInfo> findPeers(
```

Comment:
Allocation on each call? :c

Reply:
Well, I don't see any other option. Maybe the situation can be better, if we use `queue<const common::PeerInfo &>`, but for each call that queue object is a new one.